src/main/java/org/sonatype/repository/conan/internal/metadata/ConanAbsoluteUrlIndexer.java

```diff
@@ -0,0 +1,203 @@
+import org.sonatype.nexus.repository.view.payloads.StringPayload;
+import org.sonatype.nexus.transaction.UnitOfWork;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Supplier;
+
+import static java.nio.charset.Charset.defaultCharset;
+import static java.util.Collections.emptyMap;
+import static org.sonatype.nexus.repository.storage.AssetEntityAdapter.P_ASSET_KIND;
+import static org.sonatype.nexus.repository.view.Content.maintainLastModified;
+import static org.sonatype.nexus.repository.view.ContentTypes.APPLICATION_JSON;
+import static org.sonatype.repository.conan.internal.AssetKind.CONAN_SRC;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.HASH_ALGORITHMS;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.findAsset;
+import static org.sonatype.repository.conan.internal.proxy.ConanProxyHelper.toContent;
+
+/**
+ * download_url files contain absolute paths to each asset
+ *
+ * This class removes the absolute address so as to redirect back to this repository
+ *
+ * @since 3.conan
+ */
+@Singleton
+@Named
+public class ConanAbsoluteUrlIndexer
+    extends ComponentSupport
+{
+  private static final ObjectMapper MAPPER = new ObjectMapper();
+
+  @Nullable
+  public TempBlob updateAbsoluteUrls(final TempBlob tempBlob,
+                                     final Repository repository,
+                                     final String assetName) {
+    Map<String, URL> currentURLMap = readIndex(tempBlob.get(), assetName);
+    Map<String, URL> indexes = new HashMap<>();
+
+    for (Map.Entry<String, URL> entry : currentURLMap.entrySet()) {
+      URL originalUrl = entry.getValue();
+      URL indexUrl = getIndexedUrl(repository.getUrl(), entry.getValue().getPath());
+      indexes.put(entry.getValue().getPath(), originalUrl);
+      entry.setValue(indexUrl);
+    }
+
+    handleUpdatingIndexes(assetName, indexes, repository);
+
+    try {
+      return convertFileToTempBlob(MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(currentURLMap), repository);
+    }
+    catch (JsonProcessingException e) {
+      log.warn("Unable to write to blob", e);
+      return null;
+    }
+  }
+
+  private URL getIndexedUrl(final String localHost, final String path) {
+    try {
+      return new URL(localHost + path);
+    }
+    catch (MalformedURLException e) {
+      log.error("Unable to create indexed url", e);
+    }
+    return null;
+  }
+
+  private TempBlob convertFileToTempBlob(final String resolvedMap, final Repository repository) {
+    StorageFacet storageFacet = repository.facet(StorageFacet.class);
+    return storageFacet.createTempBlob(new StringPayload(resolvedMap, defaultCharset(), null), HASH_ALGORITHMS);
+  }
+
+  @TransactionalStoreMetadata
+  protected void handleUpdatingIndexes(final String assetName,
```

Comment:
Note: Looking at this I think we should be fine for concurrency as the retry mechanism will handle it for up to 8 competing requests for the same downloads_url file. Might be worth double checking manually though.